src/System.IO.FileSystem.AccessControl/src/System/IO/FileSystemAclExtensions.cs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+// Ported from DirectoryInfo.cs and made extension methods to allow extending
+// the class without System.IO.FileSystem needing to rely on System.Security.AccessControl
+
+using System;
+using System.Security.AccessControl;
+using System.Diagnostics.Contracts;
+using Microsoft.Win32.SafeHandles;
+
+namespace System.IO
+{
+    public static class FileSystemAclExtensions
+    {
+        public static DirectorySecurity GetAccessControl(this DirectoryInfo directoryInfo)
+        {
+            return new DirectorySecurity(directoryInfo.FullName, AccessControlSections.Access | AccessControlSections.Owner | AccessControlSections.Group);
+        }
+
+        public static DirectorySecurity GetAccessControl(this DirectoryInfo directoryInfo, AccessControlSections includeSections)
+        {
+            return new DirectorySecurity(directoryInfo.FullName, includeSections);
+        }
+
+        public static void SetAccessControl(this DirectoryInfo directoryInfo, DirectorySecurity directorySecurity)
+        {
+            if (directorySecurity == null)
+                throw new ArgumentNullException("directorySecurity");
+            Contract.EndContractBlock();
+
+            String fullPath = Path.GetFullPath(directoryInfo.FullName);
```

Comment:
Nit: `String` -> `string` (applies for any other cases where the language keyword should be used instead of type name)